### ISSUE TITLE
Too Many Requests error when adding 2fa or updating password

### ISSUE THREAD

**Issue by MisterSeagull** [healthchecks/healthchecks#940]
I have healthchecks installed through docker using the healthchecks/healthchecks:v3.0 image.

When I attempt to update my email address, password, or add 2fa I get an error that says too many requests.

![HealthChecks — Mozilla Firefox](https://github.com/healthchecks/healthchecks/assets/12835264/2d458bca-6711-4a58-93c1-3af2cb994b78)

![too many requests — Mozill](https://github.com/healthchecks/healthchecks/assets/12835264/8ae29f3f-50e2-4ce7-b7b6-a43bcd2d0a29)

I can edit my password through the admin management interface but I cannot setup 2fa from there. I do not get this error anywhere else. It only appears when clicking the four circled buttons.

Logs seem to indicate that the page was loaded successfully. The IP address is the internal docker IP of my reverse proxy. I can access any other links, including Django admin with no issues.

`[pid: 13|app: 0|req: 10/56] 172.25.0.99 () {64 vars in 1160 bytes} [Tue Jan 16 00:14:21 2024] GET /accounts/set_password/ => generated 1823 bytes in 389 msecs (HTTP/1.1 200) 9 headers in 399 bytes (1 switches on core 0)`

Could this be a reverse proxy problem? I am using nginx proxy manager with the default configuration.

```
server {
  set $forward_scheme http;
  set $server         "healthchecks-web-1";
  set $port           8000;

  listen 80;
listen [::]:80;

listen 443 ssl http2;
listen [::]:443 ssl http2;


  server_name hc.domain.tld;


  # Let's Encrypt SSL
  include conf.d/include/letsencrypt-acme-challenge.conf;
  include conf.d/include/ssl-ciphers.conf;
  ssl_certificate /etc/letsencrypt/live/npm-4/fullchain.pem;
  ssl_certificate_key /etc/letsencrypt/live/npm-4/privkey.pem;


    # Force SSL
    include conf.d/include/force-ssl.conf;


proxy_set_header Upgrade $http_upgrade;
proxy_set_header Connection $http_connection;
proxy_http_version 1.1;


  access_log /data/logs/proxy-host-11_access.log proxy;
  error_log /data/logs/proxy-host-11_error.log warn;


  location / {
    
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection $http_connection;
    proxy_http_version 1.1;
    

    # Proxy!
    include conf.d/include/proxy.conf;
  }


  # Custom
  include /data/nginx/custom/server_proxy[.]conf;
}
```

**Comment by cuu508** [comment]
For sensitive actions like changing email, changing password or changing 2FA methods, Healthchecks performs an additional verification action: it sends a six-digit code to user's email and asks the user to enter the code back into the app. This action is rate-limited to 10 verification attempts per day per user ([code](https://github.com/healthchecks/healthchecks/blob/42f88f4fb02f32b25415c5e192cca7eb6618d891/hc/api/models.py#L1266)). You should not normally run into this rate limit but if you're testing the system and logging in and out a bunch of times, you could hit it. I also sometimes do hit it during local development, when I'm manually testing these sensitive actions multiple times. What I usually do in that case is drop into the database shell and run `DELETE FROM api_tokenbucket;`. This discards the current rate limiter state. Obviously, don't do this on an actively used system running in production, but during initial setup or during local development it would be fine.

**Comment by MisterSeagull** [comment]
Thank you for your assistance. I did not initially have an outgoing SMTP server setup and triggered the limit before I was able to receive the email 2FA. After looking through issues and reading docs it seems like SMTP configuration is all but required, perhaps including a warning banner on an invalid or missing SMTP configuration could help guide new installs in the right direction and reduce requests for help. I had not planned on enabling outgoing SMTP emails until I realized it was required for full functionality.
